### PR TITLE
Improvements to chapter 19 (Commander: A Powerful and Simple Command Framework)

### PR DESCRIPTION
**Pull request by koendehondt**
Apart from several improvements, I removed a few paragraphs that were very confusing. I added inline review comments to indicate the paragraphs I removed.

Open issues:
* `testRemoveContact` under "Adding the Remove Contact command" is not a good test. The initial items are not specified and more importantly, the test does not verify that the selected contact has been removed. It just checks that there is one contact less than initially.
* Under "Enabling shortcuts", it reads "At the time of writing this chapter, Commander management of shortcuts has not been pushed to Spec to avoid dependency on Commander.". Is that still correct? I did not check that yet.
* This chapter mentions toolbars, but it does not add a toolbar to a window.

### CODE REVIEW

File: Chapters/Commander2/Commander.md [hunk 1]
```diff
@@ -4,188 +4,163 @@
...
+Every action is implemented as a separate command class \(subclass of `CmCommand`\) with an `execute` method and the state required for execution.
 
 ![A simple command and its hierarchy.](figures/BasicCommand.pdf width=35&label=first)
 
-We will show later that for UI framework, we need more information such as an icon and shortcut description.
-In addition, we will present how commands can be decorated with extra functionality in an extensible way.
-
-Note that nothing prevents you from defining commands by creating instances and filling them up with the same information.
-If you do so, you cannot use classes to dispatch and should put in place other mechanisms. 
-In this chapter, we only discuss how to define classes representing commands and use them.
```
Comment (koendehondt) [inline Chapters/Commander2/Commander.md:34]: This was a very confusing paragraph.

File: Chapters/Commander2/Commander.md [hunk 2]
```diff
@@ -4,188 +4,163 @@
...
-menu and menubar.
-
+A command is a simple object instance of a subclass of the class `CmCommand`. It has a description, a name \(this name can be either static or dynamic as we will show later on\). In addition, it has a context from which it extracts information to execute itself. In its basic form, there is no more than that.
 
-Note that we will present how Commander supports Spec menu and menubar creations.
-However such functionalities are not in the core of Commander.
-We show them because first, this is important to illustrate how to build user interface elements with Commander but also 
-because such functionalities show that Commander can be extended in a way that end-users do not have to feel 
-they are using special extensions.
-We will come back to such a point in the last chapter of this book to show potential extenders of Commander that they can get inspiration from the Spec extensions.
```
Comment (koendehondt) [inline Chapters/Commander2/Commander.md:53]: This was a very confusing paragraph.

File: Chapters/Commander2/Commander.md [hunk 3]
```diff
@@ -294,68 +261,53 @@ EgContactBookPresenter class >> buildContextualMenuGroupWith: presenter
...
-EgChangePhoneCommand >> newPhone: anObject
-    newPhone := anObject
-```
-
+ChangePhoneCommand >> initialize
 
-```
-EgChangePhoneCommand >> newPhone
-    ^ newPhone 
-```
```
Comment (koendehondt) [inline Chapters/Commander2/Commander.md:354]: These methods are not used.

File: Chapters/Commander2/Commander.md (hunk 1 above)
Comment (Ducasse) [reply Chapters/Commander2/Commander.md:34]: ok by me.

**Comment by Ducasse** [comment]
It would be good to push the changes to the booklet commander

**Comment by Ducasse** [comment]
May be we should move the class >> defaultSpec to the instance side

**Comment by koendehondt** [comment]
Do you refer to this change?
```
EgContactBookPresenter class >> defaultSpec
```
to
```
ContactBookPresenter class >> defaultLayout
```
which should have been 🙄
```
ContactBookPresenter >> defaultLayout
```
I will fix that.

**Comment by Ducasse** [comment]
Yes normally we have move the spec to the class and indeed we should check all the spec and turn them into layout.

**Comment by koendehondt** [comment]
> ContactBookPresenter class >> defaultLayout
>I will fix that.

Fixed in https://github.com/SquareBracketAssociates/BuildingApplicationWithSpec2/commit/e86678d165b05632b9f34babe3df63652a7aecdf

**Pull request merged**